docs/api/vega-wallet/reference/browser-extension/javascript-api.md

```diff
@@ -0,0 +1,260 @@
+public keys that appear in the `await vega.listKeys()` list are
+available for the dApp to propose transactions with:
+
+```js
+const res = await vega.sendTransaction({
+  sendingMode: 'TYPE_SYNC', // one of TYPE_ASYNC, TYPE_SYNC, TYPE_COMMIT. See below
+  publicKey: '...', // One of the keys from `listKeys()`
+  transaction: { /* ... */ }
+})
+```
+
+`sendingMode` decides when the Vega node returns a response to the transaction
+submission. `TYPE_ASYNC` will run validation checks by the network in cases 
+where this is possible, but will return fast. `TYPE_SYNC` is the perferred mode.
+This mode will return once the transaction has passed validation checks and 
+entered the mempool.  `MODE_COMMIT` is only available for debug purposes and 
+will require your own node with this mode enabled. This will cause the 
+submission to wait until the transaction has been mined.
+
+`publicKey` must be a key the user has made available to the dApp, otherwise
+the wallet will error with unknown public key.
+
+Transactions in Vega are protocol buffers. The browser wallet provides a JSON
+interface which translates to protobuf fields. As a general rule fields are
+camel-cased, falsy values are not encoded (they map to protobuf default values)
+and optional values can be set to `null` to signify explicit absence.
+
+Below are sample transactions and links to their corresponding `.d.ts` files and
+protobuf comments.
+
+#### Transfer
+
+Example below shows how to make an immediate one-off transfer from the signing
+vega keys' general account to another keys' general account.
+
+- [Type Definition](https://github.com/vegaprotocol/js-protos/blob/main/dist/vega/commands/v1/Transfer.d.ts)
+- [Protobuf definition](../../../grpc/vega/commands/v1/commands.proto.mdx#transfer)
+
+```js
+{
+  // string | number; One of the account types
+  fromAccountType: 'ACCOUNT_TYPE_GENERAL',
+  // string; hex encoded vega public key
+  to: '',
+  toAccounType: 'ACCOUNT_TYPE_GENERAL',
+  // string; $VEGA (fairground) asset id
+  asset: 'fc7fd956078fb1fc9db5c19b88f0874c4299b2a7639ad05a47a28c0aef291b55',
+  // string; 1 $VEGA
+  amount: (1n * 18n).toString(),
+  // string; optional reference
+  reference: 'My first transfer!',
+  // Make this a one-off transfer
+  kind: { 
+    oneOff: {
+      // bigint (u64); Optional delivery time in nanoseconds since UNIX epoch
+      // defaults to 0 (deliver now)
+      deliveryOn: 0n
+    } 
+  }
+}
+```
+
+#### Delegate Stake
+
+Example below shows how to delegate 1 $VEGA from the signing vega key to
+the validator Marvin on Fairground.
+
+- [Type Definition](https://github.com/vegaprotocol/js-protos/blob/main/dist/vega/commands/v1/DelegateSubmission.d.ts)
+- [Protobuf Definition](../../../grpc/vega/commands/v1/commands.proto.mdx#delegatesubmission)
+
+```js
+{
+  // string; Stake to Marvin on Fairground
+  nodeId: 'bb1822715aa86ce0e205aa4c78e9b71cdeaec94596ce72d366f0d50589eb1bf5',
+  // string; 1 $VEGA
+  amount: (1n * 18n).toString()
```

Comment:
@candida-d do we have any existing docs about how numbers are strings in vega to prevent flop errors? Can cross link to that here?

Reply:
We do! However they were published after this branch was created, so they won't be available without a rebase/merge from main.